File: languages/dotnet/runtime/Revaly.Sdk/Transport/RapHeaders.cs
namespace Revaly.Sdk.Transport;

/// <summary>Wire header names used by the runtime.</summary>
public static class RapHeaders
{
    /// <summary>The API version pin header (spec: selectable contract version).</summary>
    public const string ApiVersion = "X-Api-Version";

    /// <summary>
    /// The request correlation header. The platform echoes it on every response
    /// (generating one when the caller sent none); quote it in support tickets — it
    /// joins directly to RAP-core telemetry (DX contract §c). Note: this response
    /// header is platform middleware behaviour not yet documented in the gated spec —
    /// tracked as a platform spec follow-up.
    /// </summary>
    public const string CorrelationId = "X-Correlation-ID";
}

File: languages/php/runtime/src/Transport/RapHeaders.php
<?php

declare(strict_types=1);

namespace Revaly\Sdk\Transport;

/**
 * Header names and auth scheme used across the runtime. The RAP auth scheme is
 * `Authorization: ApiKey <key>` — the `ApiKey` prefix is mandatory (it is NOT a
 * Bearer scheme).
 */
final class RapHeaders
{
    public const AUTHORIZATION = 'Authorization';

    /** The mandatory RAP auth scheme token: `Authorization: ApiKey <key>`. */
    public const AUTH_SCHEME = 'ApiKey';

    public const USER_AGENT = 'User-Agent';

    /** Contract-version pin, sent on every request (runtime-tdd §1). */
    public const API_VERSION = 'X-Api-Version';

    /** Echoed on every response; joins merchant tickets to RAP-core telemetry. */
    public const CORRELATION_ID = 'X-Correlation-ID';
}
